### PR TITLE
drop wsTunnel and implement tunnel with ws

### PR DESCRIPTION
**Pull request by itajaja**

### CODE REVIEW

File: src/tunnel-client.js [hunk 1]
```diff
@@ -1,10 +1,65 @@
...
+      tcpConn.once('connect', () => {
+        info('TCP connection established');
+        tcpConn.on('drain', () => info('TCP connection drain'));
+        tcpConn.on('end', () => info('TCP connection end'));
+        tcpConn.on('error', () => info('TCP connection error'));
+        tcpConn.on('close', () => this._connections[identifier] = undefined);
+        tcpConn.on('data', chunk => {
+          info('TCP connection data');
+          chunk = encode(chunk, identifier);
+          this._wsConn.send(chunk, { binary: true, mask: true });
```
Comment (cancan101) [inline src/tunnel-client.js]: What is `mask` true doing?

Comment (itajaja) [inline src/tunnel-client.js]: http://stackoverflow.com/questions/14174184/what-is-the-mask-in-a-websocket-frame

File: src/tunnel-client.js [hunk 2]
```diff
@@ -1,10 +1,65 @@
...
+    info('ws message received');
+    const { chunk, identifier } = decode(data);
+    this._openConnection(identifier, (tcpConn) => tcpConn.write(chunk));
+  }
+
+  /**
+   * open a reverse tunnel on `serverUrl` from port `tunnelPort` pointing to
+   * `targetAddress`
+   */
+  open(tunnelPort) {
```
Comment (cancan101) [inline src/tunnel-client.js]: why do we round trip `tunnelPort` ?

File: src/decoder.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const ID_LENGTH = 4;
+const NETLOC_LENGTH = 20;
```
Comment (cancan101) [inline src/decoder.js]: 21

File: src/tunnel-client.js [hunk 4]
```diff
@@ -1,10 +1,51 @@
...
+
+      tcpConn.once('connect', () => {
+        info('TCP connection established');
+        tcpConn.on('drain', () => info('TCP connection drain'));
+        tcpConn.on('end', () => info('TCP connection end'));
+        tcpConn.on('error', () => info('TCP connection error'));
+        tcpConn.on('close', () => this._connections[identifier] = undefined);
+        tcpConn.on('data', tcpChunk => {
+          info('TCP connection data');
+          tcpChunk = encode(tcpChunk, identifier, ' ');
```
Comment (cancan101) [inline src/tunnel-client.js]: what is third argument?

File: src/tunnel-client.js [hunk 5]
```diff
@@ -1,10 +1,68 @@
...
+
+export default class TunnelClient {
+
+  constructor(ws) {
+    this._wsConn = ws;
+    this._connections = {};
+  }
+
+  _sendBack(buffer, identifier) {
+    const chunk = encode(buffer, identifier);
```
Comment (cancan101) [inline src/tunnel-client.js]: move this inside `if`

**Pull request merged**